gammapy/modeling/models/spectral.py

```diff
@@ -857,6 +858,89 @@ def evaluate(energy, index1, index2, amplitude, ebreak, reference, beta):
         return pwl * brk
 
 
+class PiecewiseBrokenPowerLawNormSpectralModel(SpectralModel):
+    """ Piecewise broken power-law spectral model
+       with renormalization at fixed energy nodes.
+       
+       For more information see :ref:`piecewise-broken-powerlaw-norm-spectral`.
+
+    Parameters
+    ----------
+    energy : `~astropy.units.Quantity`
+        Array of energies at which the model values are given (nodes).
+    norms : array
+        Array with the initial norms of the model at energies ``energy``.
+        A normalisation parameters is created for each value.
+        Default is one at each node.
+    """
+
+    tag = ["PiecewiseBrokenPowerLawNormSpectralModel", "pbpl-norm"]
+
+    def __init__(self, energy, norms=None):
+        self._energy = energy
+        if norms is None:
+            norms = np.ones(len(energy))
+        if len(norms) != len(energy):
+            raise ValueError("dimension mismatch")
+        if len(norms) < 2:
+            raise ValueError("Input arrays must contians at least 2 elements")
+        parameters = Parameters(
+            [Parameter(f"norm{k}", norm) for k, norm in enumerate(norms)]
+        )
+        self._set_default_parameters(parameters)
+
+    @classmethod
+    def from_parameters(cls, parameters, **kwargs):
+        norms = [p.value for p in parameters]
+        model = cls(kwargs["energy"], norms)
+        model._set_default_parameters(parameters)
+        return model
+
+    def _set_default_parameters(self, parameters):
+        for parameter in parameters:
+            setattr(self, parameter.name, parameter)
+        self.default_parameters = parameters
+
+    @property
+    def energy(self):
+        return self._energy
+
+    @property
+    def norms(self):
+        return [p.value for p in self.parameters] * u.Unit("")
+
+    def __call__(self, energy):
+        return self.evaluate(energy)
+
+    def evaluate(self, energy):
+        loge_data = np.log10(np.atleast_1d(energy.value))
+        loge = np.log10(self.energy.to(energy.unit).value)
+        logv = np.log10(self.norms)
+        log_interp = 10 ** np.interp(loge_data, loge, logv)
+        return log_interp
+
+    def to_dict(self):
+        return {
+            "type": self.tag[0],
+            "parameters": self.parameters.to_dict(),
+            "energy": {
+                "data": self.energy.data.tolist(),
+                "unit": str(self.energy.unit),
+            },
+            "norms": {"data": self.norms.data.tolist(), "unit": str(self.norms.unit),},
```

Comment:
The duplication of the norm values, here doesn't make sense to me. I think it's rather confusing then it helps (I can see how it might be shorter to define a list of norms, but it misses the extra information...). I would remove this....